### PR TITLE
Updated metadata and serviceMesh field

### PR DESCRIPTION
**Pull request by dhruv0000**
Signed-off-by: dhruv0000 <patel.4@iitj.ac.in>

**Notes for Reviewers**
1. Added Service Protos to define any abstract service under `/service`. It includes:
     - info.proto
     - health.proto
     - error.proto
2. `protos/service_mesh.proto` changes:
     - Changed `ServiceMeshes` message to `ServiceMesh` as it defines one instance of mesh.
     - Added `string version` in `ServiceMesh` message
     - Added `labels` and `annotations` in `ServiceMesh` message to define a mesh more comprehensively.
3. Updated CI workflow. Courtesy of @kumarabd

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: protos/service_mesh.proto [hunk 1]
```diff
@@ -11,7 +11,7 @@ option objc_class_prefix = "smp";
 option php_namespace = "Smp";
 option ruby_package = "Smp";
 
-message ServiceMeshes {
+message ServiceMesh {
```
Comment (dhruv0000) [inline protos/service_mesh.proto:14]: @kumarabd Any thoughts on changing `ServicesMeshes` to `ServiceMesh`. Am still conflicting with the idea.

Comment (leecalcote) [reply protos/service_mesh.proto:14]: Good reference - https://developers.google.com/protocol-buffers/docs/style

Comment (kumarabd) [reply protos/service_mesh.proto:14]: ServiceMesh it is

File: protos/metadata.proto [hunk 2]
```diff
@@ -37,3 +37,29 @@ message TrafficMetadata {
   string user_agent = 6;
   string ping_health_endpoint = 7;
 }
+
+//Controller Health
+enum ControllerStatus {
```
Comment (kumarabd) [inline protos/metadata.proto:42]: @dhruv0000  ControllerStatus, ControllerHealth, ControllerInfo, CommonError to meshkit. Just the service mesh metadata changes in here.

**Review by kumarabd — APPROVED** [review]

**Review by leecalcote — APPROVED** [review]

**Comment by dhruv0000** [comment]
@leecalcote This PR will require a minor update in `meshery` repo to use `serviceMesh` instead of `serviceMeshes` to run the performance test correctly. Please do ping me when we update the release tag, will raise that PR in `meshery`

**Comment by leecalcote** [comment]
Thanks for pointing this out, @dhruv0000. Here we are - https://github.com/layer5io/service-mesh-performance/releases/tag/v0.3.2

**Comment by dhruv0000** [comment]
[#2297](https://github.com/layer5io/meshery/pull/2297)

**Pull request merged**